dev/com.ibm.ws.transport.http/resources/com/ibm/ws/http/channel/internal/resources/httpchannelmessages.nlsprops

```diff
@@ -185,3 +185,11 @@ trusted.header.origin.invalid.value.useraction=Ensure that configured IP address
 trusted.sensitive.header.origin.invalid.value=CWWKT0039W: The [{0}] configured trustedSensitiveHeaderOrigin value is invalid and cannot be used.
 trusted.sensitive.header.origin.invalid.value.explanation=The specified value must be either a full IP address or correctly formatted hostname.
 trusted.sensitive.header.origin.invalid.value.useraction=Ensure that configured IP addresses contain a value for each segment, and that hostnames do not contain any invalid characters.
+
+config.hsts.invalid.max.age.value=CWWKT0040W: The configured max-age value [{0}] is invalid and cannot be used. The max-age value must be a number equal to or larger than 0.
+config.hsts.invalid.max.age.value.explanation=The specified max-age value must be the number 0 or a larger number.
+config.hsts.invalid.max.age.value.useraction=Ensure that the configured max-age directive contains a value of 0 or greater.
+
+config.hsts.missing.max.age=CWWKT0041W: An attempt was made to configure [{0}] as the Strict-Transport-Security header. This configuration does not contain the mandatory max-age directive. This is invalid and cannot be used.
+config.hsts.missing.max.age.explanation=The specified value must contain a valid max-age directive.
+config.hsts.missing.max.age.useraction=Ensure that the configured value contains a valid max-age directive.
```

Comment:
is there any way to give an indication of what constitutes a valid max-age directive?

Reply:
From the RFC specification, the max-age follows the same rules as `delta-second production`. Looking at another RFC that specifies the constraints of this requirement: 

> 
> 3.3.2 Delta Seconds
> 
>    Some HTTP header fields allow a time value to be specified as an
>    integer number of seconds, represented in decimal, after the time
>    that the message was received.
> 
>        delta-seconds  = 1*DIGIT

The specification section on max-age directive specifies: 

> 6.1.1.  The max-age Directive
> 
>    The REQUIRED "max-age" directive specifies the number of seconds,
>    after the reception of the STS header field, during which the UA
>    regards the host (from whom the message was received) as a Known HSTS
>    Host.  See also Section 8.1.1 ("Noting an HSTS Host - Storage
>    Model").  The delta-seconds production is specified in [RFC2616].
> 
>    The syntax of the max-age directive's REQUIRED value (after
>    quoted-string unescaping, if necessary) is defined as:
> 
>     max-age-value = delta-seconds
> 
>     delta-seconds = <1*DIGIT, defined in [RFC2616], Section 3.3.2>
> 
>    NOTE:  A max-age value of zero (i.e., "max-age=0") signals the UA to
>           cease regarding the host as a Known HSTS Host, including the
>           includeSubDomains directive (if asserted for that HSTS Host).
>           See also Section 8.1 ("Strict-Transport-Security Response
>           Header Field Processing").

The take away being that a max-age directive is considered valid when it is a non-negative integer.